Marlin/SanityCheck.h

```diff
@@ -292,7 +292,15 @@
   #endif
 
   /**
+   * Check if ENDSTOPS_ONLY_FOR_HOMING is disabled when M211 command is enabled
+   */
+  #if (defined(ENABLE_M211)) && (!defined(ENDSTOPS_ONLY_FOR_HOMING))
+    #error ENDSTOPS_ONLY_FOR_HOMING must be enabled when M211 command is active.
+  #endif
+
+  /**
    * Warnings for old configurations
+   * Check if ENDSTOPS_ONLY_FOR_HOMING is disabled when M211 command is enabled
```

Comment:
Why is it necessary to disable the soft end stops when you move the origin
to match the work-piece? They should still apply to the machine coordinates
after they are offset from the model coordinates as they are the physical
limits of the machine, which don't change, just as physical ones don't.

For example, my CNC mill running LinuxCNC has endstop switches at one end
of each axis and a soft limit to stop it crashing at the other end. They
always apply but I can touch off the work-piece and change the origin to
anywhere. The limits still apply so I cant crash an axis although I can of
course crash the tool.

Possibly Marlin applies the soft end stops are the wrong place but I think
this is the wrong way to fix it.


On 7 April 2015 at 02:06, Scott Lahteine <notifications@github.com> wrote:

> In Marlin/SanityCheck.h
> <https://github.com/MarlinFirmware/Marlin/pull/1814#discussion_r27845035>:
>
> >     * Warnings for old configurations
> > +   * Check if ENDSTOPS_ONLY_FOR_HOMING is disabled when M211 command is enabled
>
> Oops, this line is a repeat of the one above.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/MarlinFirmware/Marlin/pull/1814/files#r27845035>.
>